File: UserService/src/main/java/com/example/demo/entities/Doctor.java
package com.example.demo.entities;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "doctor")
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer doctorId;

    @ManyToOne
    @JoinColumn(name = "uid", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "specialization_id", nullable = false)
    private Specialization specialization;

    @Column(name = "base_qualification", nullable = false)
    private String baseQualification;

    @Column(name = "post_qualification")
    private String postQualification;

    private Integer experience;

    @Column(name = "consultation_fee", nullable = false)
    private BigDecimal consultationFee;

 // getters & setters
	public Integer getDoctorId() {
		return doctorId;
	}

	public void setDoctorId(Integer doctorId) {
		this.doctorId = doctorId;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Specialization getSpecialization() {
		return specialization;
	}

	public void setSpecialization(Specialization specialization) {
		this.specialization = specialization;
	}

	public String getBaseQualification() {
		return baseQualification;
	}

	public void setBaseQualification(String baseQualification) {
		this.baseQualification = baseQualification;
	}

	public String getPostQualification() {
		return postQualification;
	}

	public void setPostQualification(String postQualification) {
		this.postQualification = postQualification;
	}

	public Integer getExperience() {
		return experience;
	}

	public void setExperience(Integer experience) {
		this.experience = experience;
	}

	public BigDecimal getConsultationFee() {
		return consultationFee;
	}

	public void setConsultationFee(BigDecimal consultationFee) {
		this.consultationFee = consultationFee;
	}

    
    
}

File: Restapis/MedicalRecordsService/MedicalRecordsService/Models/Doctor.cs
using System;
using System.Collections.Generic;

namespace MedicalRecordsService.Models;

public partial class Doctor
{
    public int DoctorId { get; set; }

    public int Uid { get; set; }

    public int SpecializationId { get; set; }

    public string BaseQualification { get; set; } = null!;

    public string? PostQualification { get; set; }

    /// <summary>
    /// Years
    /// </summary>
    public int? Experience { get; set; }

    public decimal ConsultationFee { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual ICollection<DoctorAvailability> DoctorAvailabilities { get; set; } = new List<DoctorAvailability>();

    public virtual Specialization Specialization { get; set; } = null!;

    public virtual User UidNavigation { get; set; } = null!;
}
